src/js/common/parser.js

```diff
@@ -0,0 +1,223 @@
+/*
```

Comment:
We should talk offline and record some notes here.  I am already using this with front end components, and am already using the scaffolding I consciously built to provide meaningful feedback (though textual) in the right context.